imap_processing/idex/l1/idex_l1.py

```diff
@@ -45,420 +41,75 @@ class Scitype(IntEnum):
-        ),
-        TriggerDescription(
-            "reflectron_voltage",
-            "IDX__TXHDRHVPSHKCH3",
-            12,
-            "Reflectron Voltage",
-            (
-                "Last measurement in raw dN for processor board signal: "
-                "Reflectron Voltage"
-            ),
-            "Voltage",
-            "dN",
-        ),
-        TriggerDescription(
-            "rejection_voltage",
-            "IDX__TXHDRHVPSHKCH4",
-            12,
-            "Rejection Voltage",
-            (
-                "Last measurement in raw dN for processor board signal: "
-                "Rejection Voltage"
-            ),
-            "Voltage",
-            "dN",
-        ),
-        TriggerDescription(
-            "detector_current",
-            "IDX__TXHDRHVPSHKCH5",
-            12,
-            "Detector Current",
-            (
-                "Last measurement in raw dN for processor board signal: "
-                "Detector Current "
-            ),
-            "Current",
-            "dN",
-        ),
+        TriggerDescription("event_number", "IDX__TXHDREVTNUM"),
+        TriggerDescription("tof_high_trigger_level", "IDX__TXHDRHGTRIGLVL"),
+        TriggerDescription("tof_high_trigger_num_max_1_2", "IDX__TXHDRHGTRIGNMAX12"),
+        TriggerDescription("tof_high_trigger_num_min_1_2", "IDX__TXHDRHGTRIGNMIN12"),
+        TriggerDescription("tof_high_trigger_num_min_1", "IDX__TXHDRHGTRIGNMIN1"),
+        TriggerDescription("tof_high_trigger_num_max_1", "IDX__TXHDRHGTRIGNMAX1"),
+        TriggerDescription("tof_high_trigger_num_min_2", "IDX__TXHDRHGTRIGNMIN2"),
+        TriggerDescription("tof_high_trigger_num_max_2", "IDX__TXHDRHGTRIGNMAX2"),
+        TriggerDescription("tof_low_trigger_level", "IDX__TXHDRLGTRIGLVL"),
+        TriggerDescription("tof_low_trigger_num_max_1_2", "IDX__TXHDRLGTRIGNMAX12"),
+        TriggerDescription("tof_low_trigger_num_min_1_2", "IDX__TXHDRLGTRIGNMIN12"),
+        TriggerDescription("tof_low_trigger_num_min_1", "IDX__TXHDRLGTRIGNMIN1"),
+        TriggerDescription("tof_low_trigger_num_max_1", "IDX__TXHDRLGTRIGNMAX1"),
+        TriggerDescription("tof_low_trigger_num_min_2", "IDX__TXHDRLGTRIGNMIN2"),
+        TriggerDescription("tof_low_trigger_num_max_2", "IDX__TXHDRLGTRIGNMAX2"),
+        TriggerDescription("tof_mid_trigger_level", "IDX__TXHDRMGTRIGLVL"),
+        TriggerDescription("tof_mid_trigger_num_max_1_2", "IDX__TXHDRMGTRIGNMAX12"),
+        TriggerDescription("tof_mid_trigger_num_min_1_2", "IDX__TXHDRMGTRIGNMIN12"),
+        TriggerDescription("tof_mid_trigger_num_min_1", "IDX__TXHDRMGTRIGNMIN1"),
+        TriggerDescription("tof_mid_trigger_num_max_1", "IDX__TXHDRMGTRIGNMAX1"),
+        TriggerDescription("tof_mid_trigger_num_min_2", "IDX__TXHDRMGTRIGNMIN2"),
+        TriggerDescription("tof_mid_trigger_num_max_2", "IDX__TXHDRMGTRIGNMAX2"),
+        TriggerDescription(
+            "low_sample_coincidence_mode_blocks", "IDX__TXHDRLSTRIGCMBLOCKS"
+        ),
```

Comment:
You could probably throw a `# noqa` at the end of this line so that `ruff` doesn't auto-correct this. I think its fine that this line exceeds the line limit for the sake of being consistent with the other lines.

Reply:
Ah! That would be great! For some reason, # noqa is not  working here... Is there something else I could try?

Reply:
I was playing around with this and its a bit finicky, but I got this to work. Needed to add `#fmt: skip` as well, I guess to tell `ruff-format` check to ignore the line specifically

```python
trigger_description_dict = {
    trigger.name: trigger
    for trigger in [
        TriggerDescription("event_number", "IDX__TXHDREVTNUM"),
        TriggerDescription("tof_high_trigger_level", "IDX__TXHDRHGTRIGLVL"),
        TriggerDescription("tof_high_trigger_num_max_1_2", "IDX__TXHDRHGTRIGNMAX12"),
        TriggerDescription("tof_high_trigger_num_min_1_2", "IDX__TXHDRHGTRIGNMIN12"),
        TriggerDescription("tof_high_trigger_num_min_1", "IDX__TXHDRHGTRIGNMIN1"),
        TriggerDescription("tof_high_trigger_num_max_1", "IDX__TXHDRHGTRIGNMAX1"),
        TriggerDescription("tof_high_trigger_num_min_2", "IDX__TXHDRHGTRIGNMIN2"),
        TriggerDescription("tof_high_trigger_num_max_2", "IDX__TXHDRHGTRIGNMAX2"),
        TriggerDescription("tof_low_trigger_level", "IDX__TXHDRLGTRIGLVL"),
        TriggerDescription("tof_low_trigger_num_max_1_2", "IDX__TXHDRLGTRIGNMAX12"),
        TriggerDescription("tof_low_trigger_num_min_1_2", "IDX__TXHDRLGTRIGNMIN12"),
        TriggerDescription("tof_low_trigger_num_min_1", "IDX__TXHDRLGTRIGNMIN1"),
        TriggerDescription("tof_low_trigger_num_max_1", "IDX__TXHDRLGTRIGNMAX1"),
        TriggerDescription("tof_low_trigger_num_min_2", "IDX__TXHDRLGTRIGNMIN2"),
        TriggerDescription("tof_low_trigger_num_max_2", "IDX__TXHDRLGTRIGNMAX2"),
        TriggerDescription("tof_mid_trigger_level", "IDX__TXHDRMGTRIGLVL"),
        TriggerDescription("tof_mid_trigger_num_max_1_2", "IDX__TXHDRMGTRIGNMAX12"),
        TriggerDescription("tof_mid_trigger_num_min_1_2", "IDX__TXHDRMGTRIGNMIN12"),
        TriggerDescription("tof_mid_trigger_num_min_1", "IDX__TXHDRMGTRIGNMIN1"),
        TriggerDescription("tof_mid_trigger_num_max_1", "IDX__TXHDRMGTRIGNMAX1"),
        TriggerDescription("tof_mid_trigger_num_min_2", "IDX__TXHDRMGTRIGNMIN2"),
        TriggerDescription("tof_mid_trigger_num_max_2", "IDX__TXHDRMGTRIGNMAX2"),
        TriggerDescription("low_sample_coincidence_mode_blocks", "IDX__TXHDRLSTRIGCMBLOCKS"),  # noqa
        TriggerDescription("low_sample_trigger_polarity", "IDX__TXHDRLSTRIGPOL"),
        TriggerDescription("low_sample_trigger_level", "IDX__TXHDRLSTRIGLVL"),
        TriggerDescription("low_sample_trigger_num_min", "IDX__TXHDRLSTRIGNMIN"),
        TriggerDescription("low_sample_trigger_mode", "IDX__TXHDRLSTRIGMODE"),
        TriggerDescription("tof_low_trigger_mode", "IDX__TXHDRLSTRIGMODE"),
        TriggerDescription("tof_mid_trigger_mode", "IDX__TXHDRMGTRIGMODE"),
        TriggerDescription("tof_high_trigger_mode", "IDX__TXHDRHGTRIGMODE"),
        TriggerDescription("detector_voltage", "IDX__TXHDRHVPSHKCH0"),
        TriggerDescription("sensor_voltage", "IDX__TXHDRHVPSHKCH1"),
        TriggerDescription("target_voltage", "IDX__TXHDRHVPSHKCH2"),
        TriggerDescription("reflectron_voltage", "IDX__TXHDRHVPSHKCH3"),
        TriggerDescription("rejection_voltage", "IDX__TXHDRHVPSHKCH4"),
        TriggerDescription("detector_current", "IDX__TXHDRHVPSHKCH5"),
    ]
}  # fmt: skip
```